modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js

```diff
@@ -934,6 +934,10 @@ return view.extend({
 				o = ss.taboption('general', CBIWifiFrequencyValue, '_freq', '<br />' + _('Operating frequency'));
 				o.ucisection = s.section;
 
+				o = ss.taboption('general', form.Value, 'channels', _('Limit auto scan channels'),
+					_('Space seperated or list range of channels with a hyphen i.e. 100 140 34-100'));
+				o.optional = true;
```

Comment:
Please add a validation function for this option.

Reply:
Will adding

`o.datatype = 'uinteger';`

I would think making sure numbers are added only? or what a good other type to use?

Reply:
I might have used a naïve regex along the lines of `[\d- ]+` to limit the accepted character types in this field, and let hostapd deal with what's usable.

Reply:
Review other code examples, or test that this works:
```js
		o.validate = function(section_id, value) {
			const regex = /[\d -]+/;
			if (!regex.test(value)) {
				return _('Error: %s; expecting space separated numbers or hyphenated number ranges.').format(value);
			}
			return true;
		}
```

Reply:
> and let hostapd deal with what's usable.

Does it "deal" with unusable values or will it just refuse to start? In either case the regex above is incomplete. The validation function should check for proper syntax, e.g. no unfinished ranges (`1-` or `-12`, ensure that start value is smaller than end value etc.)

Reply:
I just tested it again with broken ranges and it starts fine (broken ranges are ignored, and the currently chosen working channel is added into `chanlist` if no other valid ranges were found in `chanlist`). In this case, there is no need for ad absurdum checks (although they don't hurt to have), although your improved regex is also good.

Reply:
Extended testing:

While invalid ranges don't prevent hostapd startup, they propagate into the config, and likely won't be useful when it comes time to change channels. hostapd seems to stop processing chanlist after the first invalid range.

Space separated ranges: 
`        option channels '36-40 44-48 56-60'` results in `chanlist=36-40 44-48 56-60`

Comma separated ranges: 

`        option channels '36-40,44-48,56-60'` results in `chanlist=36-40,44-48,56-60`

Space+comma separated ranges:

`        option channels '36-40, 44-48, 56-60'` results in `chanlist=36-40, 44-48, 56-60`

Invalid ranges: processing stops after first range

`        option channels 36-, 44-, 56-60'` results in `chanlist=36-,`


Summary: using a widget here might not be a bad idea.

Reply:
Please test that this works:
```js

		o.validate = function(section_id, value) {
			/* Regular expression to match valid value pattern
			* 36-40 44-48 56-60
			* 36-40,44-48,56-60
			* 36-40, 44-48, 56-60
			* 36, 44, 56-60
			* Fail: 36-, 44-, 56-60
			*/
			var regex = /^(\d+-\d+|\d+)(\s*,\s*(\d+-\d+|\d+))*$/;

			// Check if value matches the pattern
			if (!regex.test(value)) {
				return _('Error: %s; expecting space or comma separated numbers or hyphenated number ranges.').format(value);
			}

			// Split value by commas or spaces
			var ranges = value.split(/\s*,\s*|\s+/);

			// Validate each range
			for(var range of ranges) {
				// If the range has a '-' character
				if (range.includes("-")) {
					var [start, end] = range.split("-").map(Number);
					// Check that start and end are valid integers and start is less than or equal to end
					if (isNaN(start) || isNaN(end) || start > end) {
						return _('Error: %s; expecting space or comma separated numbers or hyphenated number ranges.').format(value);
					}
				} else { // It's a single integer
					if (isNaN(parseInt(range)))
						return _('Error: %s; expecting space or comma separated numbers or hyphenated number ranges.').format(value);
				}
			}
			return true;
		}

```